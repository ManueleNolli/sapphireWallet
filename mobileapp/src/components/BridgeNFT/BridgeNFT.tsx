import React from 'react'
import { Button, Layout, Spinner, StyleService, Text, useStyleSheet } from '@ui-kitten/components'
import { vh, vw } from '../../Styles'
import useBridgeNFT from './useBridgeNFT'
import { ScrollView, TouchableOpacity, View } from 'react-native'
import { NFTCard } from '../NFTCard/NFTCard'
import { OwnedNFT } from '../../services/blockchain'

type SendETHProps = {
  address: string
  close: () => void
}
export default function BridgeNFT({ address, close }: SendETHProps) {
  const styles = useStyleSheet(themedStyles)
  const { isSendLoading, isNFTLoading, sendBridgeTransaction, nfts, selectedNFT, setSelectedNFT } = useBridgeNFT({
    address,
    close,
  })

  const LoadingIndicator = (props: any) => {
    return (
      isSendLoading && (
        <View style={[props.style, styles.indicator]}>
          <Spinner size="tiny" />
        </View>
      )
    )
  }

  return (
    <Layout style={styles.container}>
      <Text
        category="h6"
        style={{
          paddingTop: 2 * vh,
          paddingHorizontal: 2 * vw,
        }}
      >
        Bridge NFT
      </Text>
      <Text
        style={{
          marginBottom: 2 * vh,
          paddingHorizontal: 2 * vw,
        }}
      >
        Select an NFT to bridge
      </Text>
      {isNFTLoading && (
        <View
          style={{
            marginTop: 2 * vh,
          }}
        >
          <Spinner
            size="giant"
            status="basic"
            style={{
              marginBottom: 2 * vh,
            }}
          />
        </View>
      )}
      {nfts.length === 0 && !isNFTLoading && (
        <View style={{ marginTop: 2 * vh }}>
          <Text category="label">No NFTs found</Text>
        </View>
      )}
      <ScrollView style={{ marginTop: 2 * vh }} showsVerticalScrollIndicator={false}>
        {nfts.map((nft: OwnedNFT, index: number) => (
          <TouchableOpacity
            role="button"
            key={index}
            style={[styles.nftContainer, index == selectedNFT ? styles.nftSelected : null]}
            onPress={() => setSelectedNFT(index)}
          >
            <NFTCard
              name={nft.name}
              tokenId={nft.tokenId}
              collectionName={nft.collectionName}
              collectionDescription={nft.collectionDescription}
              image={nft.image}
              network={nft.network}
              style={{
                height: 35 * vh,
                width: 35 * vh,
              }}
            />
          </TouchableOpacity>
        ))}
      </ScrollView>
      <Button
        testID="send-button"
        accessoryRight={<LoadingIndicator />}
        style={{ marginTop: 2 * vh, width: '100%' }}
        appearance="outline"
        status="info"
        disabled={nfts.length === 0}
        onPress={sendBridgeTransaction}
      >
        Bridge NFT
      </Button>
    </Layout>
  )
}

const themedStyles = StyleService.create({
  container: {
    paddingVertical: 2 * vh,
    paddingHorizontal: 2 * vw,
    alignItems: 'center',
    borderRadius: 10,
    maxHeight: 90 * vh,
    width: 90 * vw,
  },
  indicator: {
    position: 'absolute',
    right: 5,
    alignItems: 'center',
  },
  nftContainer: {
    height: 80 * vw,
    width: 80 * vw,
    justifyContent: 'center',
    alignItems: 'center',
    borderRadius: 10,
  },
  nftSelected: {
    backgroundColor: 'color-primary-400',
    height: 76 * vw,
    width: 76 * vw,
    marginLeft: 2 * vw,
  },
})
