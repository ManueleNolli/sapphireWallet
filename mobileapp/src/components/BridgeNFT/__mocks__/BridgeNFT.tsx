const BridgeNFTMock = () => <div data-testid="BridgeNFT">BridgeNFT</div>

export default BridgeNFTMock
