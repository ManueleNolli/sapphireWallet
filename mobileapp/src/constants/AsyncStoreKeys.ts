const theme = 'theme'
const firstAccess = 'firstAccess'

const EOAAddress = 'EOAAddress'

const walletContractAddress = 'walletContractAddress'

export { theme, firstAccess, EOAAddress, walletContractAddress }
