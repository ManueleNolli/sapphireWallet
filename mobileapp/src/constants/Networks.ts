export enum NETWORKS {
  LOCALHOST = 'Localhost',
  SEPOLIA = 'Sepolia',
}
