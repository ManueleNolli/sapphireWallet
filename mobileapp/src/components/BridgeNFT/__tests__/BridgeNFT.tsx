import { NETWORKS } from '../../../constants/Networks'
import { NFTPlaceholder } from '../../../assets/AssetsRegistry'
import renderWithTheme from '../../../TestHelper'
import useBridgeNFT from '../useBridgeNFT'
import { fireEvent, waitFor } from '@testing-library/react-native'
import React from 'react'
import BridgeNFT from '../BridgeNFT'

jest.mock('../useBridgeNFT', () => jest.fn())
jest.mock('../../InputAddress/InputAddress')
jest.mock('../../InputNumeric/InputNumeric')
jest.mock('../../QRCodeScanner/QRCodeScanner')

jest.mock('@ui-kitten/components', () => {
  const { View } = require('react-native')
  return {
    ...jest.requireActual('@ui-kitten/components'),
    Spinner: () => <View data-testid="spinner" />,
  }
})

describe('BridgeNFT', () => {
  beforeEach(() => {
    jest.clearAllMocks()
  })
  it('render correctly', () => {
    ;(useBridgeNFT as jest.Mock).mockReturnValue({
      isSendLoading: false,
      isNFTLoading: false,
      nfts: [
        {
          name: 'name0',
          description: 'description0',
          image: NFTPlaceholder,
          tokenId: 0,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress0',
          collectionName: 'collectionName0',
          collectionDescription: 'collectionDescription0',
        },
        {
          name: 'name1',
          description: 'description1',
          image: NFTPlaceholder,
          tokenId: 1,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress1',
          collectionName: 'collectionName1',
          collectionDescription: 'collectionDescription1',
        },
      ],
    })

    const tree = renderWithTheme(<BridgeNFT address="address" close={() => {}} />)

    expect(tree).toMatchSnapshot()
  })

  it('render correctly without nfts', () => {
    ;(useBridgeNFT as jest.Mock).mockReturnValue({
      isLoading: false,
      isNFTLoading: false,
      nfts: [],
    })

    const tree = renderWithTheme(<BridgeNFT address="address" close={() => {}} />)

    expect(tree.getAllByText('No NFTs found')).toBeTruthy()
    expect(tree).toMatchSnapshot()
  })

  it('render correctly loading', () => {
    ;(useBridgeNFT as jest.Mock).mockReturnValue({
      isLoading: false,
      isNFTLoading: true,
      nfts: [],
    })

    const tree = renderWithTheme(<BridgeNFT address="address" close={() => {}} />)
    expect(tree).toMatchSnapshot()
  })

  it('Select NFT will change', () => {
    const setSelectedNFT = jest.fn((number: number) => {})
    ;(useBridgeNFT as jest.Mock).mockReturnValue({
      isLoading: false,
      isNFTLoading: true,
      setSelectedNFT,
      nfts: [
        {
          name: 'name0',
          description: 'description0',
          image: NFTPlaceholder,
          tokenId: 0,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress0',
          collectionName: 'collectionName0',
          collectionDescription: 'collectionDescription0',
        },
        {
          name: 'name1',
          description: 'description1',
          image: NFTPlaceholder,
          tokenId: 1,
          network: NETWORKS.LOCALHOST,
          collectionAddress: 'collectionAddress1',
          collectionName: 'collectionName1',
          collectionDescription: 'collectionDescription1',
        },
      ],
    })

    const tree = renderWithTheme(<BridgeNFT address="address" close={() => {}} />)

    const NFTCard = tree.getByTestId('NFTCard-1')

    fireEvent.press(NFTCard)

    expect(setSelectedNFT).toHaveBeenCalledWith(1)
  })
})
