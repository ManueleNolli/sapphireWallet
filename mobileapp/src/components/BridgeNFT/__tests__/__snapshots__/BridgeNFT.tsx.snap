// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BridgeNFT render correctly 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1200.6,
        "paddingHorizontal": 15,
        "paddingVertical": 26.68,
        "width": 675,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "paddingHorizontal": 15,
          "paddingTop": 26.68,
        },
      ]
    }
  >
    Bridge NFT
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "marginBottom": 26.68,
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    Select an NFT to bridge
  </Text>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      {
        "marginTop": 26.68,
      }
    }
  >
    <View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 600,
            "justifyContent": "center",
            "opacity": 1,
            "width": 600,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "red",
                "borderColor": "#598BFF",
                "borderRadius": 10,
                "borderWidth": 1,
                "elevation": 5,
                "height": 667,
                "overflow": "hidden",
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 10,
                "width": "100%",
              },
              {
                "height": 466.9,
                "width": 466.9,
              },
            ]
          }
          testID="NFTCard-0"
        >
          <View
            style={
              {
                "backgroundColor": "red",
                "height": "102%",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "top": -2,
                "width": "100%",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              bottom={0}
              contentFit="cover"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              left={0}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              position="absolute"
              right={0}
              source={[]}
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              top={0}
              transition={null}
            />
            <View
              style={
                {
                  "alignItems": "flex-end",
                  "marginHorizontal": 15,
                  "marginVertical": 26.68,
                }
              }
            >
              <ViewManagerAdapter_ExpoImage
                contentFit="contain"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                height={36}
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={[]}
                style={
                  {
                    "height": 36,
                    "width": 36,
                  }
                }
                transition={null}
                width={36}
              />
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "borderRadius": 8,
                  "marginHorizontal": 15,
                  "marginVertical": 26.68,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="CollapsibleButton"
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxHeight": 600.3,
                      "paddingHorizontal": 15,
                      "paddingVertical": 13.34,
                    },
                  ]
                }
              >
                <ViewManagerAdapter_ExpoBlurView
                  blurReductionFactor={4}
                  intensity={50}
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  tint="default"
                />
                <View
                  style={
                    {
                      "width": "90%",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="h4"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 26,
                            "fontWeight": "800",
                          },
                          undefined,
                        ]
                      }
                    >
                      name0
                    </Text>
                    <Text
                      appearance="default"
                      category="label"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 12,
                            "fontWeight": "800",
                          },
                          {
                            "marginRight": 7.5,
                          },
                        ]
                      }
                    >
                      #0
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "height": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      onLayout={[Function]}
                      style={{}}
                    >
                      <View>
                        <Text
                          appearance="default"
                          category="h6"
                          status="basic"
                          style={
                            [
                              {
                                "color": "#222B45",
                                "fontFamily": "System",
                                "fontSize": 18,
                                "fontWeight": "800",
                              },
                              undefined,
                            ]
                          }
                        >
                          collectionName0
                        </Text>
                        <Text
                          appearance="default"
                          category="p1"
                          status="basic"
                          style={
                            [
                              {
                                "color": "#222B45",
                                "fontFamily": "System",
                                "fontSize": 15,
                                "fontWeight": "400",
                              },
                              undefined,
                            ]
                          }
                        >
                          collectionDescription0
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "height": "100%",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "transform": [
                          {
                            "scale": 1,
                          },
                        ],
                      }
                    }
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="32"
                      bbWidth="32"
                      fill="#222B45"
                      focusable={false}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          {
                            "height": 32,
                            "tintColor": "#222B45",
                            "width": 32,
                          },
                          {
                            "flex": 0,
                            "height": 32,
                            "width": 32,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                    >
                      <RNSVGGroup
                        fill={
                          {
                            "payload": 4280429381,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGGroup
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGRect
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              height="24"
                              matrix={
                                [
                                  -1,
                                  1.2246467991473532e-16,
                                  -1.2246467991473532e-16,
                                  -1,
                                  24,
                                  24,
                                ]
                              }
                              opacity={0}
                              width="24"
                              x="0"
                              y="0"
                            />
                            <RNSVGPath
                              d="M18 15a1 1 0 0 1-.64-.23L12 10.29l-5.37 4.32a1 1 0 0 1-1.41-.15 1 1 0 0 1 .15-1.41l6-4.83a1 1 0 0 1 1.27 0l6 5a1 1 0 0 1 .13 1.41A1 1 0 0 1 18 15z"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "alignItems": "center",
            "borderRadius": 10,
            "height": 600,
            "justifyContent": "center",
            "opacity": 1,
            "width": 600,
          }
        }
      >
        <View
          style={
            [
              {
                "backgroundColor": "red",
                "borderColor": "#598BFF",
                "borderRadius": 10,
                "borderWidth": 1,
                "elevation": 5,
                "height": 667,
                "overflow": "hidden",
                "shadowColor": "black",
                "shadowOffset": {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 1,
                "shadowRadius": 10,
                "width": "100%",
              },
              {
                "height": 466.9,
                "width": 466.9,
              },
            ]
          }
          testID="NFTCard-1"
        >
          <View
            style={
              {
                "backgroundColor": "red",
                "height": "102%",
                "justifyContent": "space-between",
                "overflow": "hidden",
                "top": -2,
                "width": "100%",
              }
            }
          >
            <ViewManagerAdapter_ExpoImage
              bottom={0}
              contentFit="cover"
              contentPosition={
                {
                  "left": "50%",
                  "top": "50%",
                }
              }
              left={0}
              onError={[Function]}
              onLoad={[Function]}
              onLoadStart={[Function]}
              onProgress={[Function]}
              placeholder={[]}
              position="absolute"
              right={0}
              source={[]}
              style={
                {
                  "bottom": 0,
                  "left": 0,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                }
              }
              top={0}
              transition={null}
            />
            <View
              style={
                {
                  "alignItems": "flex-end",
                  "marginHorizontal": 15,
                  "marginVertical": 26.68,
                }
              }
            >
              <ViewManagerAdapter_ExpoImage
                contentFit="contain"
                contentPosition={
                  {
                    "left": "50%",
                    "top": "50%",
                  }
                }
                height={36}
                onError={[Function]}
                onLoad={[Function]}
                onLoadStart={[Function]}
                onProgress={[Function]}
                placeholder={[]}
                source={[]}
                style={
                  {
                    "height": 36,
                    "width": 36,
                  }
                }
                transition={null}
                width={36}
              />
            </View>
            <View
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onClick={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                {
                  "borderRadius": 8,
                  "marginHorizontal": 15,
                  "marginVertical": 26.68,
                  "opacity": 1,
                  "overflow": "hidden",
                }
              }
              testID="CollapsibleButton"
            >
              <View
                style={
                  [
                    {
                      "backgroundColor": "transparent",
                    },
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                      "justifyContent": "space-between",
                      "maxHeight": 600.3,
                      "paddingHorizontal": 15,
                      "paddingVertical": 13.34,
                    },
                  ]
                }
              >
                <ViewManagerAdapter_ExpoBlurView
                  blurReductionFactor={4}
                  intensity={50}
                  style={
                    {
                      "bottom": 0,
                      "left": 0,
                      "position": "absolute",
                      "right": 0,
                      "top": 0,
                    }
                  }
                  tint="default"
                />
                <View
                  style={
                    {
                      "width": "90%",
                    }
                  }
                >
                  <View
                    style={
                      {
                        "alignItems": "center",
                        "flexDirection": "row",
                        "justifyContent": "space-between",
                      }
                    }
                  >
                    <Text
                      appearance="default"
                      category="h4"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 26,
                            "fontWeight": "800",
                          },
                          undefined,
                        ]
                      }
                    >
                      name1
                    </Text>
                    <Text
                      appearance="default"
                      category="label"
                      status="basic"
                      style={
                        [
                          {
                            "color": "#222B45",
                            "fontFamily": "System",
                            "fontSize": 12,
                            "fontWeight": "800",
                          },
                          {
                            "marginRight": 7.5,
                          },
                        ]
                      }
                    >
                      #1
                    </Text>
                  </View>
                  <View
                    collapsable={false}
                    pointerEvents="none"
                    style={
                      {
                        "height": 0,
                        "overflow": "hidden",
                      }
                    }
                  >
                    <View
                      collapsable={false}
                      onLayout={[Function]}
                      style={{}}
                    >
                      <View>
                        <Text
                          appearance="default"
                          category="h6"
                          status="basic"
                          style={
                            [
                              {
                                "color": "#222B45",
                                "fontFamily": "System",
                                "fontSize": 18,
                                "fontWeight": "800",
                              },
                              undefined,
                            ]
                          }
                        >
                          collectionName1
                        </Text>
                        <Text
                          appearance="default"
                          category="p1"
                          status="basic"
                          style={
                            [
                              {
                                "color": "#222B45",
                                "fontFamily": "System",
                                "fontSize": 15,
                                "fontWeight": "400",
                              },
                              undefined,
                            ]
                          }
                        >
                          collectionDescription1
                        </Text>
                      </View>
                    </View>
                  </View>
                </View>
                <View
                  style={
                    {
                      "height": "100%",
                    }
                  }
                >
                  <View
                    collapsable={false}
                    style={
                      {
                        "transform": [
                          {
                            "scale": 1,
                          },
                        ],
                      }
                    }
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="32"
                      bbWidth="32"
                      fill="#222B45"
                      focusable={false}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          {
                            "height": 32,
                            "tintColor": "#222B45",
                            "width": 32,
                          },
                          {
                            "flex": 0,
                            "height": 32,
                            "width": 32,
                          },
                        ]
                      }
                      vbHeight={24}
                      vbWidth={24}
                    >
                      <RNSVGGroup
                        fill={
                          {
                            "payload": 4280429381,
                            "type": 0,
                          }
                        }
                        propList={
                          [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGGroup
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGGroup
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                          >
                            <RNSVGRect
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                              height="24"
                              matrix={
                                [
                                  -1,
                                  1.2246467991473532e-16,
                                  -1.2246467991473532e-16,
                                  -1,
                                  24,
                                  24,
                                ]
                              }
                              opacity={0}
                              width="24"
                              x="0"
                              y="0"
                            />
                            <RNSVGPath
                              d="M18 15a1 1 0 0 1-.64-.23L12 10.29l-5.37 4.32a1 1 0 0 1-1.41-.15 1 1 0 0 1 .15-1.41l6-4.83a1 1 0 0 1 1.27 0l6 5a1 1 0 0 1 .13 1.41A1 1 0 0 1 18 15z"
                              fill={
                                {
                                  "payload": 4278190080,
                                  "type": 0,
                                }
                              }
                            />
                          </RNSVGGroup>
                        </RNSVGGroup>
                      </RNSVGGroup>
                    </RNSVGSvgView>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": false,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 149, 255, 0.08)",
        "borderColor": "#0095FF",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 26.68,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
    testID="send-button"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "#0095FF",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Bridge NFT
    </Text>
  </View>
</View>
`;

exports[`BridgeNFT render correctly loading 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1200.6,
        "paddingHorizontal": 15,
        "paddingVertical": 26.68,
        "width": 675,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "paddingHorizontal": 15,
          "paddingTop": 26.68,
        },
      ]
    }
  >
    Bridge NFT
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "marginBottom": 26.68,
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    Select an NFT to bridge
  </Text>
  <View
    style={
      {
        "marginTop": 26.68,
      }
    }
  >
    <View
      data-testid="spinner"
    />
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      {
        "marginTop": 26.68,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(143, 155, 179, 0.16)",
        "borderColor": "rgba(143, 155, 179, 0.24)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 26.68,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
    testID="send-button"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "rgba(143, 155, 179, 0.48)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Bridge NFT
    </Text>
  </View>
</View>
`;

exports[`BridgeNFT render correctly without nfts 1`] = `
<View
  appearance="default"
  level="1"
  style={
    [
      {
        "backgroundColor": "#FFFFFF",
      },
      {
        "alignItems": "center",
        "borderRadius": 10,
        "maxHeight": 1200.6,
        "paddingHorizontal": 15,
        "paddingVertical": 26.68,
        "width": 675,
      },
    ]
  }
>
  <Text
    appearance="default"
    category="h6"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 18,
          "fontWeight": "800",
        },
        {
          "paddingHorizontal": 15,
          "paddingTop": 26.68,
        },
      ]
    }
  >
    Bridge NFT
  </Text>
  <Text
    appearance="default"
    category="p1"
    status="basic"
    style={
      [
        {
          "color": "#222B45",
          "fontFamily": "System",
          "fontSize": 15,
          "fontWeight": "400",
        },
        {
          "marginBottom": 26.68,
          "paddingHorizontal": 15,
        },
      ]
    }
  >
    Select an NFT to bridge
  </Text>
  <View
    style={
      {
        "marginTop": 26.68,
      }
    }
  >
    <Text
      appearance="default"
      category="label"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 12,
            "fontWeight": "800",
          },
          undefined,
        ]
      }
    >
      No NFTs found
    </Text>
  </View>
  <RCTScrollView
    showsVerticalScrollIndicator={false}
    style={
      {
        "marginTop": 26.68,
      }
    }
  >
    <View />
  </RCTScrollView>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": true,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": "rgba(143, 155, 179, 0.16)",
        "borderColor": "rgba(143, 155, 179, 0.24)",
        "borderRadius": 4,
        "borderWidth": 1,
        "flexDirection": "row",
        "justifyContent": "center",
        "marginTop": 26.68,
        "minHeight": 40,
        "minWidth": 40,
        "opacity": 1,
        "paddingHorizontal": 10,
        "paddingVertical": 12,
        "width": "100%",
      }
    }
    testID="send-button"
  >
    <Text
      appearance="default"
      category="p1"
      status="basic"
      style={
        [
          {
            "color": "#222B45",
            "fontFamily": "System",
            "fontSize": 15,
            "fontWeight": "400",
          },
          {
            "color": "rgba(143, 155, 179, 0.48)",
            "fontFamily": "System",
            "fontSize": 14,
            "fontWeight": "bold",
            "marginHorizontal": 10,
          },
        ]
      }
    >
      Bridge NFT
    </Text>
  </View>
</View>
`;
